### ISSUE TITLE
Fix `function-no-unknown` false positive for `scroll()`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#6537]
### What steps are needed to reproduce the bug?

use [`scroll()`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-timeline/scroll)
with `layout.css.scroll-linked-animations.enabled` enabled

### Question

My question is more general; should we preemptively fix stuff or should we wait for issues to pop up once it's not behind a flag anymore?
i.e. preemptive measures or late fixes

ref
- https://www.w3.org/TR/scroll-animations-1/#scroll-notation
- https://caniuse.com/mdn-css_properties_animation-timeline_scroll

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-no-unknown": "true"
  }
}
```

### Proposal to fix the bug

maybe PR https://github.com/niksy/css-functions-list/

**Comment by ybiquitous** [comment]
@Mouvedia Good catch. I think we can do both:

- open an issue or pull request to [niksy/css-functions-list](https://github.com/niksy/css-functions-list)
- add arbitrary notations to the `functionsList` array below with a `TODO` comment including the issue or pull request above:
  https://github.com/stylelint/stylelint/blob/e30ec863f8bc1d5ed87c23e3c5a554a01bc4350b/lib/rules/function-no-unknown/index.js#L46

If a new version of `css-functions-list` is released soon, the latter is unnecessary.

**Comment by ybiquitous** [comment]
I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by Mouvedia** [comment]
> Please consider [contributing](https://stylelint.io/contributing) if you have time.

Alright, Ill give it a try.

**Comment by ybiquitous** [comment]
We can reproduce a false positive for `scroll()` on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilTMEsBbJAF3wkwFozCYAbfTGeFAZzACcI66AKLiCQCUAbiwBfLCAA0IVgE9MJJAA8E4VqxngKAM3wBzdRmwp0IDgFc6MVuZYmcOc7suYwZCpUwRKbgNY+AO6Y9igkVjBYOJKY4iDiQA):

```css
a {
  animation-timeline: scroll(root);
                   /* ↑ Unexpected unknown function "scroll" */
}
```

```json
{
  "rules": {
    "function-no-unknown": true
  }
}
```

@Mouvedia Right?

**Comment by Mouvedia** [comment]
> Right?

Indeed, check the PR tests.

**Comment by ybiquitous** [comment]
FYI. I've opened https://github.com/niksy/css-functions-list/pull/5 to update the function list.
